metis/lib/server/controllers/bucket_controller.rb

```diff
@@ -77,29 +77,25 @@ def find
     query = Metis::Query.new(
       project_name: @params[:project_name],
       bucket: bucket,
-      params: params
+      params: params,
+      limit: limit,
+      offset: offset,
     )
 
     results = query.execute
 
     files = results[:files]
     folders = results[:folders]
 
-    limit = limit ? limit : [files.length, folders.length].max
```

Comment:
limit and offset no longer applied at this leve, but in the query object instead.